tests/integration/test_alembic_migrations.py

```diff
@@ -1,16 +1,28 @@
 import pathlib
 import time
 
+import pytest
 from alembic import command
 from alembic.config import Config
-from flaky import flaky
 from sqlalchemy import Column, Integer, String, create_engine, text
 from sqlalchemy.orm import declarative_base, sessionmaker
 
 wait_lsn = text("select pglogical.wait_slot_confirm_lsn(NULL, NULL)")
 
 
-@flaky(max_runs=5, min_passes=2)
+def wait_for_query_truthiness(sessionmaker_, query, truthy=True):
+    for _ in range(30):
+        # make the session inside the loop to make sure we're not holding any cursors, etc.
+        with sessionmaker_() as session:
+            session.execute(wait_lsn)
+            maybe_results = session.execute(query)
+            result = maybe_results.first()
+            if (result is not None) == truthy:
+                break
+    return result
+
+
+@pytest.mark.flaky(max_runs=3)
```

Comment:
**question (non-blocking)**: Why are these tests flaky?

Reply:
sometimes WAL takes several minutes to ship - I *think* this is due to the low transaction volume, but I'm not sure. Through trial-and-error I found that retrying on short timeouts was faster than waiting longer once